File: Data-Structures/generic/subsetBinary.java
class subsetBinary{

    static void binary(int digits,String prefix){
        // System.out.println("digits value : " + digits + " " + "prefix value : " + prefix);
        if(digits==0){ 
            System.out.println(prefix);
        }
        else{
            binary(digits-1, prefix + "0");
            binary(digits-1, prefix + "1");
        }
    }

    public static void main(String[] args) {
        binary(3,"");
    }
}

File: Data-Structures/generic/subsetBinary.cpp
#include <iostream>
using namespace std;

void subsetBinary(int digits,string prefix=""){
    if(digits==0){
        cout<<prefix<<endl;
    }
    else{
        subsetBinary(digits-1,prefix+"0");
        subsetBinary(digits-1,prefix+"1");
    }
}

int main(){
    subsetBinary(3);
    return 0;
}